Update OAuth docs to use domain parameter (#21316)
DataDog/documentation: content/en/developers/authorization/oauth2_in_datadog.md

   - This parameter is only provided if the user initiates authorization from the Datadog integration tile. See the [Initiate authorization from a third-party location](#Initiate-authorization-from-a-third-party-location) section for more options if the user chooses to initiate authorization externally.  
   - The `domain` query parameter provides the [Datadog site][17] that the authorizing user is in, and is required to construct the URL for this GET request to the Authorize endpoint: `https://api.<domain>/oauth2/v1/authorize?...`.

> <div class="alert alert-info">The `domain` parameter was introduced in July 2023, and replaced the previous `site` parameter. The `site` parameter is still backwards compatible, but new OAuth clients should only use the `domain` parameter moving forward.  Store this value in a secure database or location for later use with additional API endpoints.</div>

Reviewer: ```suggestion
<div class="alert alert-info">The `domain` parameter was introduced in July 2023, and replaced the previous `site` parameter. The `site` parameter is still backwards compatible, but new OAuth clients should only use the `domain` parameter moving forward. Store this value in a secure database or location for later use with additional API endpoints.</div>
```

Additionally, when users initiate authorization from the Datadog integration tile, they are required to have corresponding permissions for all requested scopes. If authorization is initiated from somewhere other than the integration tile, users without all of the required permissions may complete authorization (but are prompted to re-authorize with proper permissions when they return to the Datadog integration tile). 

Datadog recommends that partners prompt users to initiate authorization from Datadog, rather than from their own platform.

While Datadog does not recommend supporting initiating authorization from a third-party location anywhere outside of the Datadog integration tile, if you do choose to go down this path then you must ensure that you are able to support users in all Datadog sites, and are willing to continue to support new Datadog sites as they may be created. This usually includes implementing a way for the user to manually input their site onto your platform while authorizing. 

> Keep in mind that organizations may have subdomains as well (i.e. https://subdomain.datadgohq.com). Subdomains should not be included in API calls. To ensure that users are authorizing in the correct site, always direct them to the US1 Datadog site (`app.datadoghq.com`), and from there, they can select their region.

Reviewer: ```suggestion
Keep in mind that organizations may have subdomains as well (for example, https://subdomain.datadoghq.com). Do not include subdomains in API calls. To ensure that users are authorizing in the correct site, always direct them to the US1 Datadog site (`app.datadoghq.com`), and from there, they can select their region.
```
